mobile/android/android-application-manager.ts

```diff
@@ -44,10 +44,16 @@ export class AndroidApplicationManager extends ApplicationManagerBase {
 	}
 
 	public async startApplication(appIdentifier: string): Promise<void> {
-		await this.adb.executeShellCommand(["monkey",
-			"-p", appIdentifier,
-			"-c", "android.intent.category.LAUNCHER",
-			"1"]);
+		const pmDumpOutput = await this.adb.executeShellCommand(["pm", "dump", appIdentifier, "|", "grep", "-A", "1", "MAIN"]);
+		const fullActivityNameRegExp = /([A-Za-z]{1}[A-Za-z\d_]*\.)*[A-Za-z][A-Za-z\d_]*\/([a-z][a-z_0-9]*\.)*[A-Z_]($[A-Z_]|[\w_])*/;
```

Comment:
Here's an example of the `pm dump | grep -A 1 MAIN`:
```
       android.intent.action.MAIN:
          3b2df03 org.nativescript.cliapp/com.tns.NativeScriptActivity filter 50dd82e
            Action: "android.intent.action.MAIN"
            Category: "android.intent.category.LAUNCHER"
--
        intent={act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=org.nativescript.cliapp/com.tns.NativeScriptActivity}
        realActivity=org.nativescript.cliapp/com.tns.NativeScriptActivity
--
            Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=org.nativescript.cliapp/com.tns.NativeScriptActivity }
            frontOfTask=true task=TaskRecord{fe592ac #449 A=org.nativescript.cliapp U=0 StackId=1 sz=1}
```
and the regex finds the full path to the activity name, which in this case is: `org.nativescript.cliapp/com.tns.NativeScriptActivity`